### ISSUE TITLE
Diverging training?

### ISSUE THREAD

**Issue by alfassy** [JiuhaiChen/BLIP3o#75]
Hi dear authors, and @JiuhaiChen,
Thank you for the great work.
I have tried to train your vanilla Blip3O with Qwen, after 1000 steps I did inference and asked for a cute cat and got images which resemble cats, then in the 2000 step I got very noisy images which don't look anything like a cat, I attach them here.

<img width="2048" height="2048" alt="Image" src="https://github.com/user-attachments/assets/bd21fccb-1219-474c-819c-4c5f1bc9a2fb" />
Looking at the loss graph I do a a spike in the loss at around 1274 steps, I attach the loss graph as well: 

<img width="618" height="449" alt="Image" src="https://github.com/user-attachments/assets/ef31a657-f117-4871-9d0f-7339a624fcc0" />

Did you also see cases like that? Any insight?
Thank you,
Amit

**Comment by alfassy** [comment]
I am looking at the deepspeed files and notice that I used zero2 instead of zero1 which was used by default and zero2 doesn't have gradient_clipping set so that might be the difference and solution here.

**Comment by alfassy** [comment]
<img width="2048" height="2048" alt="Image" src="https://github.com/user-attachments/assets/90c7f07b-ddb7-49d1-8743-b10125066e50" />

![](vscode-remote://ssh-remote%2Blogin2.bluevela.rmf.ibm.com/proj/mmfm/amit/dev/Blip_fresh/GVO/A%20photo%20of%20cute%20cat.png)
So after fixing the Stability issue, that's what I get as "a photo of a cute cat" after 31K steps using batch size of 80 over 32 gpus. Seems ok?

**Comment by JiuhaiChen** [comment]
It seems ok since some JourneyDB data are mixed, any changes if you train more steps?

**Comment by alfassy** [comment]
<img width="635" height="638" alt="Image" src="https://github.com/user-attachments/assets/8cd2f657-06b5-49ed-8295-781168deb382" />
That's after 62K steps of pretraining

**Comment by JiuhaiChen** [comment]
I feel it's too JourneyDB style, any more examples? And you can quickly try to do the instruction tuning on https://huggingface.co/datasets/BLIP3o/BLIP3o-60k.

**Comment by alfassy** [comment]
Apparently I used pretrain, pretrain-short-caption and pretrain-journrydb so short and journeydb are there twice. So that's why. Fixed now thx, I'll send update after training some more

**Comment by zzxslp** [comment]
HI @alfassy do you mind sharing your zero2.json file that works? Thanks!

**Comment by alfassy** [comment]
{
    "fp16": {
        "enabled": "auto",
        "loss_scale": 0,
        "loss_scale_window": 1000,
        "initial_scale_power": 16,
        "hysteresis": 2,
        "min_loss_scale": 1
    },
    "bf16": {
        "enabled": "auto"
    },
    "gradient_clipping": 1.0,
    "train_micro_batch_size_per_gpu": "auto",
    "train_batch_size": "auto",
    "gradient_accumulation_steps": "auto",
    "steps_per_print": 100,
    "wall_clock_breakdown": false,
    "zero_optimization": {
        "stage": 2,
        "overlap_comm": false,
        "contiguous_gradients": true,
        "sub_group_size": 1e9,
        "reduce_bucket_size": "auto"
    }
}

**Comment by zzxslp** [comment]
Hi @alfassy thank you so much! I tried zero2, but a batch size of 80 (per device) will throw OOM. I am following default set-up of training script, except changing to zero2, and also using 4 nodes so assume I can replicate your training. 
Do you use other settings to save memory? What would be the trade off for memory-speed in this case?

**Comment by alfassy** [comment]
I used 72 batch size on 80GB A100. I tried to increase learning rate with batch size but this caused training to diverge so I went back to the original learning rate. Authors said to do 150K+ steps and I didn't have the time to compare performance mid training to see if the model converged early due to the batch size, it should in theory. 
No other settings.
